DPL: improve metric sending

ktf opened the pull request.

ktf committed:
DPL: introduce DataProcessingStates to keep track of binary states

* Similar to metrics, but only tracks the most recent value for a
  given buffer, which can be used either for strings or to provide
  a relatively updated view of the state to (among other things) the
  driver.

ktf committed:
DPL: improve metric sending

ktf committed:
DPL: fix some of the metrics for online consumption

* Make sure they are not tagged with Subsystem DPL.
* Send the initial value.
* Publish the metrics every 60 seconds, even if not updated.

alibuild commented:
Error while checking build/O2/fullCI for 985bca65992e42d5f0d09ae4d81d15188a14074e at 2023-03-24 09:33:
```
## sw/BUILD/DataDistribution-latest/log
collect2: error: ld returned 1 exit status
gmake[2]: *** [src/ReadoutEmulator/CMakeFiles/ReadoutEmulator.dir/build.make:245: bin/ReadoutEmulator] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:668: src/ReadoutEmulator/CMakeFiles/ReadoutEmulator.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
gmake: *** [Makefile:146: all] Error 2


## sw/BUILD/o2checkcode-latest/log
--
++ echo '========== List of errors found =========='
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/3a595e7164b633a2bde9df9378d00c0207569674/slc8_x86-64/o2checkcode/1.0-local312/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10997/985bca65992e42d5f0d09ae4d81d15188a14074e/build_O2_fullCI/pretty.html).

ktf committed:
DPL: improve latency metrics

Still wrong due to the fact the DataProcessingHeader does not really
contain a timestamp from 1970.

alibuild commented:
Error while checking build/O2/fullCI for 045cfc459d5c09d9b965492d6912e4fa0c45b00e at 2023-03-24 15:33:
```
## sw/BUILD/O2-latest/log
CMake Error at cmake/O2AddLibrary.cmake:76 (add_library):
CMake Error at cmake/O2AddLibrary.cmake:76 (add_library):
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10997/045cfc459d5c09d9b965492d6912e4fa0c45b00e/build_O2_fullCI/pretty.html).

ktf committed:
DPL: demote alarm messages when not running online.
